bears/xml/XMLBear.py

```diff
@@ -24,10 +26,11 @@ class XMLBear(LocalBear, Lint):
     gives_corrected = True
 
     def process_output(self, output, filename, file):
-        if self.stdout_output:  # only yield Result if stdout is not empty
-            return self._process_corrected(self.stdout_output, filename, file)
-        if self.stderr_output:  # pragma: no cover
-            return self._process_issues(self.stderr_output, filename)
+        if self.stderr_output:
+            self.use_stderr = True
+        return itertools.chain(
+            self._process_issues(self.stderr_output, filename),
+            self._process_corrected(self.stdout_output, filename, file))
```

Comment:
This won't work, because you are not checking if `self.stdout_output` is empty or not. Thus if I have:

`test.xml` as:

```xml
<a>hey & hi</a>
```

and I do:

```
❯ coala --files=test.xml --bears=XMLBear -L DEBUG
```

I get:

```
test.xml
|   1| <a>hey•&•hi</a>
|    | [NORMAL] XMLBear:
|    | XML can be formatted better.
|    | The following actions are applicable to this result:
|    |  0: Apply no further actions.
|    |  1: Open the affected file(s) in an editor.
|    |  2: Apply the patch automatically.
|    |  3: Print a diff of the patch that would be applied.
|    | Please enter the number of the action you want to execute. 3
|----|    | /home/sanket/coala-bears/test.xml
|    |++++| /home/sanket/coala-bears/test.xml
|   1|    |-<a>hey & hi</a>
|    | Displayed patch successfully.
|    | The following actions are applicable to this result:
|    |  0: Apply no further actions.
|    |  1: Open the affected file(s) in an editor.
|    |  2: Apply the patch automatically.
|    |  3: Print a diff of the patch that would be applied.
|    | Please enter the number of the action you want to execute. 
```

As you can see, the diff is removing the line, because `stdout` is empty.